modules/querier/querier.go

```diff
@@ -251,34 +246,37 @@ func (q *Querier) FindTraceByID(ctx context.Context, req *tempopb.TraceByIDReque
 				return r.Get(traceKey, ring.Read, nil, nil, nil)
 			}
 		}
+		var spanCountTotal, traceCountTotal atomic.Int64
+		var found atomic.Bool
 
 		// get responses from all ingesters in parallel
 		span.AddEvent("searching ingesters")
-		responses, err := q.forIngesterRings(ctx, userID, getRSFn, func(funcCtx context.Context, client tempopb.QuerierClient) (interface{}, error) {
-			return client.FindTraceByID(funcCtx, req)
-		})
-		if err != nil {
-			return nil, fmt.Errorf("error querying ingesters in Querier.FindTraceByID: %w", err)
-		}
-
-		found := false
-		for _, r := range responses {
-			t := r.response.(*tempopb.TraceByIDResponse).Trace
+		forEach := func(funcCtx context.Context, client tempopb.QuerierClient) error {
```

Comment:
nit:  Not critical, but what do you think about making these inline funcs again like:

```go
err := q.forIngesterRings(ctx, userID, getRSFn, func(ctx, client) error {
    <callback logic here>
})
```

Reply:
I moved them out to make them easy to read because now these callback functions is few more lines. 

I personally find it easy to parse out when the callback is outside but I don't have a strong preference for either way, if you think that it's easy to read when inlined, I can inline them.